core/kazoo/src/kz_doc.erl

```diff
@@ -242,9 +250,31 @@ external_attachments(JObj, Default) ->
 attachment_names(JObj) ->
     kz_json:get_keys(attachments(JObj, kz_json:new())).
 
+-spec attachment_revision(kz_json:object()) -> pos_integer().
+attachment_revision(AttachmentJObj) ->
+    {[Values], _} = kz_json:get_values(AttachmentJObj),
+    kz_json:get_value(?ATTACHMENT_PROPERTY_REVISION, Values).
+
+-spec compare_attachments(kz_json:object(), kz_json:object()) -> boolean().
+compare_attachments(AttachmentJObjA, AttachmentJObjB) ->
+    attachment_revision(AttachmentJObjA) =< attachment_revision(AttachmentJObjB).
+
+-spec latest_attachment_id(kz_json:object()) -> ne_binary() | 'undefined'.
+latest_attachment_id(Doc) ->
```

Comment:
Do you have non-task document dumps I could use?
Or I could just move this back to kazoo_tasks…